packages/pds/src/pipethrough.ts

```diff
@@ -1,22 +1,131 @@
+import express from 'express'
 import * as ui8 from 'uint8arrays'
-import { jsonToLex } from '@atproto/lexicon'
-import { HandlerPipeThrough } from '@atproto/xrpc-server'
-import { CallOptions, ResponseType, XRPCError } from '@atproto/xrpc'
-import { lexicons } from './lexicon/lexicons'
+import net from 'node:net'
+import { LexValue, jsonToLex, stringifyLex } from '@atproto/lexicon'
+import { HandlerPipeThrough, InvalidRequestError } from '@atproto/xrpc-server'
+import { ResponseType, XRPCError } from '@atproto/xrpc'
+import { ids, lexicons } from './lexicon/lexicons'
 import { httpLogger } from './logger'
-import { noUndefinedVals } from '@atproto/common'
+import { getServiceEndpoint, noUndefinedVals } from '@atproto/common'
+import AppContext from './context'
+
+const defaultService = (
+  ctx: AppContext,
+  path: string,
+): { url: string; did: string } | null => {
+  const nsid = path.replace('/xrpc/', '')
```

Comment:
A few misc things to maybe just peek at here:
 - Do we need to check that the path starts with `/xrpc` too?
 - May want to handle trailing slash in the URL, I believe the pds routing permits it.
 - Express's `req.url` is relative to the nearest router or something like that, I forget the details but may be relevant since I know it gets passed here.  There is also `req.originalUrl` which might be more reliable for us.